src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -57,18 +56,22 @@ export default class HTTPClient implements IClient {
       this.clientUrl = `${this.clientHost}:${this.clientPort}/v1.0`;
     }
 
-    this.client = fetch;
+    if(!HTTPClient.client) {
+      HTTPClient.client = fetch;
+    }
 
     // Add a custom agent so we can decide if we want to reuse connections or not
     // we use an agent so we can reuse an open connection, limiting handshake requirements
     // Note: when using an agent, we will encounter TCPWRAP since the connection doesn't get destroyed
-    if (this.options.isKeepAlive) {
-      this.httpAgent = new http.Agent({ keepAlive: true, keepAliveMsecs: 30 * 1000 });
-      this.httpsAgent = new https.Agent({ keepAlive: true, keepAliveMsecs: 30 * 1000 });
-    } else {
-      this.httpAgent = new http.Agent();
-      this.httpsAgent = new https.Agent();
-    }
+    const keepAlive = this.options.isKeepAlive;
+    const keepAliveMsecs = 30 * 1000; // it is applicable only when keepAlive is set to true
+
+      if(!HTTPClient.httpAgent) {
```

Comment:
Resolve correct indentation please, seems to be 4 spaces instead of 2?

Reply:
My bad - corrected now.